### ISSUE TITLE
No (exect first) location is saved

### ISSUE THREAD

**Issue by sekiroh** [mauron85/cordova-plugin-background-geolocation#137]
## Your Environment

* Plugin version: cordova-plugin-mauron85-background-geolocation spec="~2.2.0-alpha.1"
* Platform: Android
* OS version: 4.4.2
* Device manufacturer and model: Huawei Y540
* Cordova version (`cordova -v`): 6.3.0
* Cordova platform version (`cordova platform ls`): android 5.2.1
* Plugin configuration options: 
```js
{   desiredAccuracy: 100,
    stationaryRadius: 30,
    distanceFilter: 1,
    interval: 10*1000,
    stopOnTerminate: false,
    activityType: 'AutomotiveNavigation',
    debug: true
  }
```

## Context

Tracking geolocation while driving for 10 km.

## Actual Behavior

Program acquires well locations, but has persisted only one of them at beginning. So callback function has been called only once. See [logfile](https://gist.github.com/sekiroh/3ba1d908ba4f609684cd305f6e6d0d6f)

**Comment by mauron85** [comment]
What is the number of locations returned by getLocations() and getValidLocations()?

**Comment by sekiroh** [comment]
Thank you for your quick reply. getLocations returns only one location if I remember. I haven't called getValidLocations.

I have just done another test, and no problem here. getLocations and getValidLocations return 20 entries. See [logfile](https://gist.github.com/sekiroh/ef633e5734f7c09b2906f0a8d4445fcf)

This time, wifi connection wasn't enabled when starting, perhaps a problem when previous location is obtained from network, and new from gps?

**Comment by mauron85** [comment]
In first logs there is mix of locations from gps and network. In second there are only locations from gps.
You can always check provider from logs:

GPS
```
Location change: Location[gps...
```

Network
```
Location change: Location[network...
```

However, there is some suspicious behaviour that's worth investigating:
```
2016/08/09 23:11:44] LocationService:Previous location: [network acc=38.0 time=1470776557324] is better than current
[2016/08/09 23:11:44] LocationService:New location BGLocation[gps 49.******,5.****** id=null acc=4 et=+11m59s843ms alt=***** vel=0.0 bear=292.46 {Bundle[mParcelledData.dataSize=40]} locprov=0]
```

meaning that location from network is better than GPS. That usually should not be happening.

**Comment by mauron85** [comment]
Can you repeat first test, with 2.2.0.alfa.3. I've added timestamp logging, so it would be possible to tell why network location is preferred over gps.

**Comment by mauron85** [comment]
Ok I've found problem in your log data.

Network provider reported timestamp: 1470776557324 which is: `2016-08-09T21:02:37.324Z`
```
[2016/08/09 23:07:26] DistanceFilterLocationProvider:Test provider=network lat=49.******2 lon=5.******6 acy=38.0 v=0.0m/s time=1470776557324
```

GPS sampling recorded time=1470773246000 which is `2016-08-09T20:07:26.000Z`
```
[2016/08/09 23:07:26] DistanceFilterLocationProvider:Test provider=gps lat=49.****** lon=5.******666666666 acy=9.7 v=22.765135m/s time=1470773246000
```

In other words GPS provider is reporting time one hour behind network provider.

**Comment by sekiroh** [comment]
Ok, I'll try to reproduce this. (I've just open another ticket which talks about gps time before seeing your reply).

**Comment by mauron85** [comment]
I'm going to change algorithm to take into account such a differences. Wait couple of minutes for .alfa.4.

**Comment by mauron85** [comment]
Fixed in 2.2.0-alpha.4. Please reopen if necessary
